### DISCUSSION TITLE
Adding new output variable of USWRFItoa

### DISCUSSION THREAD

**Discussion by XiaSun-Atmos** [ufs-community/ufs-weather-model#1671] [Q&A] [answered]
Hi,
I am trying to add a new output variable of` USWRFItoa`, instantaneous top of atmos upward shortwave flux. It is defined in GFS_diagnostics.F90. This variable is added to diag_table:
`"gfs_phys",  "USWRFItoa",     "uswrf_toa",   "fv3_history2d",         "all",  .false.,  "none",  2`
However, the output values of `uswrf_toa` from fv3 runs are in magnitudes of 1e6 W/m**2, which should be in magnitudes of 1e3 W/m**2. The spatial distribution does look very similar to `uswrf_avetoa`, which is one hour avg of `uswrf_toa`.
What should I do to make `uswrf_toa `on the right magnitudes?
Thanks,

**Comment by yangfanglin** [comment]
Does the surface DSWRFI look fine ?

There might be an issue.   All the TOA instantaneous SW and LW fluxes are passed from fluxr in ccpp/driver/GFS_diagnostics.F90; However,  fluxr in ccpp/physics/physics/GFS_rrtmg_post.F90  are accumulated variables.  They are not instantaneous fluxes.  

Surface instantaneous fluxes look to be fine.   

Bring the attention to 
@dustinswales  @grantfirl  @Qingfu-Liu @junwang-noaa

**Comment by XiaSun-Atmos** [comment]
Thanks @yangfanglin . All the other instantaneous radiation fluxes that use their designated variable names, such as `dswsfci(:)` and `ulwsfci(:)` (not `fluxr`) look good.
cc: @joeolson42

**Comment by XiaSun-Atmos** [comment]
Hi, I am following up with this discussion.
Is there a plan or timeline for fixing the code for the TOA instantaneous SW and LW fluxes?
Thanks.

**Comment by yangfanglin** [comment]
@XiaSun-Atmos , We at EMC do not have a plan to work on this yet.   It would be great if you or anyone else in the community can fix this bug.

**Comment by RatkoVasic-NOAA** [comment]
@yangfanglin Is this issue solved?

**Reply by XiaSun-Atmos** [reply, marked as answer]
I don't think so, unless some other developers have fixed it.
